### ISSUE TITLE
Propose dropping support for Node 0.10 and 0.11

### ISSUE THREAD

**Issue by markstos** [UnbounDev/node-strava-v3#57]
Node .10 and .12 have been end-of-lifed.

If the minimum  supported version was moved up to Node 4.x then we could use newer Node.js features in the docs and code like arrow functions. 

Node .10 and  .12 users would still have the option to use older versions of the module, or fork the module if they'd rather maintain a compatible version of node-strava-v3 instead of upgrading their Node version.

**Comment by TimoSta** [comment]
Hi @markstos,
I agree that support for Node.js versions should be dropped once they are out of maintenance.

In fact, the first and admittedly trivial step into this direction was already taken a few weeks ago. We used to run the Travis test suite exclusively on Node .10 and .11. I included more versions into the test suite, up to the most recent one.

In order to stick to semanitc versioning, dropping older Node versions needs to happen as part of a new major version, considering that it's a breaking change.

My proposal would be

1. to introduce a depreciation warning into a new minor version of 1.x if the library is used with Node.js .10, .12 or 5 (and potentially also 7 as its maintenance window ends in June '17), and
2. to publish a new major version 2, which allows changes that use features implemented in Node 4 and higher.

I will look into this topic a bit further over the next few days and make the necessary preparations.
-- Timo
